### PR TITLE
Remove TPM-specific API surface and dependencies from v2 SDK

### PR DESCRIPTION
**Pull request by drwill-ms**

### CODE REVIEW

File: provisioning/service/src/Enrollments/AttestationMechanismType.cs [hunk 1]
```diff
@@ -22,23 +22,14 @@ public enum AttestationMechanismType
...
-
         /// <summary>
         /// x509 attestation mechanism.
         /// </summary>
         /// <remarks>
         /// Identify the attestation mechanism as <see cref="X509Attestation"/>.
         /// </remarks>
         [EnumMember(Value = "x509")]
-        X509 = 2,
+        X509 = 1,
```
Comment (drwill-ms) [inline provisioning/service/src/Enrollments/AttestationMechanismType.cs:32]: Any concerns about renumbering here?

Comment (timtay-microsoft) [reply provisioning/service/src/Enrollments/AttestationMechanismType.cs:32]: none from me

**Comment by timtay-microsoft** [comment]
Don't forget to remove all TPM startup steps from our ADO pipelines

File: provisioning/device/src/Transports/Mqtt/ProvisioningTransportHandlerMqtt.cs [hunk 2]
```diff
@@ -366,10 +366,10 @@ private MqttClientOptionsBuilder CreateMqttClientOptions(ProvisioningTransportRe
             {
                 password = ProvisioningSasBuilder.BuildSasSignature(key1.GetPrimaryKey(), string.Concat(provisioningRequest.IdScope, '/', "registrations", '/', provisioningRequest.Authentication.GetRegistrationId()), TimeSpan.FromHours(1));
             }
-            else if (provisioningRequest.Authentication is AuthenticationProviderTpm)
-            {
-                throw new NotSupportedException("TPM authentication is not supported over MQTT TCP or MQTT web socket.");
-            }
+            //else if (provisioningRequest.Authentication is AuthenticationProviderTpm)
```
Comment (timtay-microsoft) [inline provisioning/device/src/Transports/Mqtt/ProvisioningTransportHandlerMqtt.cs:369]: Delete?

**Review by timtay-microsoft — APPROVED** [review]

File: provisioning/service/src/Enrollments/AttestationMechanism.cs [hunk 3]
```diff
@@ -27,11 +26,7 @@ internal AttestationMechanism(Attestation attestation)
                 throw new ArgumentNullException(nameof(attestation));
             }
 
-            if (attestation is TpmAttestation tpmAttestation)
-            {
-                Tpm = tpmAttestation;
-            }
-            else if (attestation is X509Attestation x509Attestation)
+             if (attestation is X509Attestation x509Attestation)
```
Comment (patilsnr) [inline provisioning/service/src/Enrollments/AttestationMechanism.cs:29]: nit: is there an extra space here?

File: provisioning/service/src/Enrollments/AttestationMechanismType.cs (hunk 1 above)
Comment (abhipsaMisra) [reply provisioning/service/src/Enrollments/AttestationMechanismType.cs:32]: no concerns

**Review by abhipsaMisra — APPROVED** [review]
We can also update our yaml script to no longer look for the tpm exe/build the tpm docker image.

**Review by tmahmood-microsoft — APPROVED** [review]

**Review by andyk-ms — COMMENTED** [review]
Do we plan to remove tpm in:
build.ps1
security/tpm
vsts/start_tpm_windows.ps1
vsts/TpmSimulator (folder)
vsts/*.yaml (tpm test docker)
vsts/*.yml (tpm test docker)

**Comment by drwill-ms** [comment]
> 

Done

**Pull request merged**